### DISCUSSION TITLE
Test code or tested code did not (only) close its own output buffers

### DISCUSSION THREAD

**Discussion by PankovAlxndr** [laravel/framework#46724] [Q&A] [answered]
PHP 8.2
Laravel 10
Pest 2

```
test('create new user', function () {
    Storage::fake('avatars');
    $image = UploadedFile::fake()->image('avatar.png');

    expect(1)->toBe(1);
});
```

Output:
`./vendor/bin/pest --filter UserCreateTest

   WARN  Tests\Feature\Users\UserCreateTest
  ! create new user → Test code or tested code did not (only) close its own output buffers   `

Why?

**Comment by k-pognerybko** [comment, marked as answer]
В UploadedFile используется метод generateImage из /vendor/laravel/framework/src/Illuminate/Http/Testing/FileFactory.php
Для его корректной работы нужен модуль GD (для imagecreatetruecolor($width, $height);)
После установки зависимости проблема должна исчезнуть

**Comment by PankovAlxndr** [comment]
Yes, everything is correct. Here, at this link https://qna.habr.com/q/1270368, you can read the solution in Russian.
